storage-proofs/porep/src/nse/vanilla/labels.rs

```diff
@@ -701,4 +837,80 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn test_gpu_cpu_consistency() {
+        let rng = &mut XorShiftRng::from_seed(crate::TEST_SEED);
```

Comment:
This test is very repetitive, which makes it hard to verify that CPU and GPU code paths are equivalent.

Can you restructure (perhaps including some of the suggested refactorings of the implementation) so it becomes obvious?

Something roughly like this would be good:

```
let test_fn = |use_gpu, …| { … };

let (a_cpu, b_cpu, c_cpu) = test_fn(false);
let (a_gpu, b_gpu, c_gpu) = test_fn(true);

assert_eq!(a_cpu, a_gpu);
assert_eq!(b_cpu, b_gpu);
assert_eq!(c_cpu, c_gpu);
```